src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/BuildUpToDateCheck.cs

```diff
@@ -143,16 +142,31 @@ void AddImports(IReadOnlyList<IProjectImportSnapshot> value)
             AddImports(e.Value);
         }
 
-        private void OnSourceItemChanged(IProjectSubscriptionUpdate e)
+        private void OnSourceItemChanged(IProjectSubscriptionUpdate e, IProjectItemSchema projectItemSchema)
         {
-            foreach (var itemType in e.ProjectChanges.Where(changes => changes.Value.Difference.AnyChanges))
+            var itemTypes = new HashSet<string>(projectItemSchema.GetKnownItemTypes().Where(itemType => projectItemSchema.GetItemType(itemType).UpToDateCheckInput));
```

Comment:
This particular line of code only iterates over the list of known item types, it isn't affected by the actual number of files. Or were you concerned about something further down?